src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1309,7 +1309,11 @@ public function registerForAutoconfiguration(string $interface)
     /**
      * Registers an attribute that will be used for autoconfiguring annotated classes.
      *
-     * The configurator will receive a ChildDefinition instance, an instance of the attribute and the corresponding \ReflectionClass, in that order.
+     * If an attribute can be used on multiple targets, a union type should be used for
+     * the \Reflector parameter. Oterwise specify one \Reflector type.
+     *
+     * @param class-string $attributeClass
+     * @param callable(ChildDefinition, object $attribute, \ReflectionClass|\ReflectionMethod|\ReflectionProperty|\ReflectionParameter):void $configurator
```

Comment:
There is an issue with this annotation: if one type hints `ReflectionClass $foo`, psalm will complain that the type hint too narrow. I'm not sure we should use this actually.

Reply:
Reverted the changes of today.

Reply:
Please don't revert this. Nicolas has a point. But we can just use `\Reflector` as typehint then and clarify in the doc that this is either `\ReflectionClass|\ReflectionMethod|\ReflectionProperty|\ReflectionParameter` or a union of them depending on the attribute target.

Reply:
@tobion Applied your feebback.